src/routes/friend.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const router = express.Router();
+const User = require("../databases/schema/users");
+
+//Send a friend request to a user and add the user to the senders list of pending requests
+//Incoming: user's object _id, username to search for
+//Outgoing: status 200/error will be empty
+router.put("/addFriend", async (request, response) => {
+    const {id, search} = request.body;
+    const userObj = await User.findById(id);
+    //search by username for the requestee, if not found, return an error
+    const friendReq = await User.findOne({username : search});
+    if(!friendReq) {
+        return response.status(404).send({error: "No user exists for that username!"});
+    }
+    await User.findByIdAndUpdate(friendReq._id, { $push: { requests: userObj.username } });
+    await User.findByIdAndUpdate(id, { $push: { pending: friendReq.username } });
+    return response.status(200).send({error: ""});
+});
+
+//Accept or reject a friend request, either way, remove the request from both sides
+//Incoming: user's object _id, username of the request sender, 1 to accept or 0 to reject
+//Outgoing: status 200/error will be empty
+router.put("/processRequest", async (request, response) => {
+    const id = request.body.id;
+    const requester = request.body.requester;
+    const accept = request.body.accept;
+
+    const userObj = await User.findById(id);
+    const newFriendObj = await User.findOne({username: requester});
+    const newFriend = newFriendObj._id;
+    //Check to see if the requester still exists
+    if(!newFriendObj) {
+        return response.status(404).send({error: "This user doesn't exist anymore!"});
+    }
+    //Remove the requester from your requests list, and remove the user from the requester's pending list
+    await User.findByIdAndUpdate(id, { $pull: { requests: requester } });
+    await User.findByIdAndUpdate(newFriend, { $pull: { pending: userObj.username } });
+    if(accept) {
+        //Add both users to their respective friend lists
+        await User.findByIdAndUpdate(id, { $push: { friends: newFriend } });
+        await User.findByIdAndUpdate(newFriend, { $push: { friends: id } });
+    }
+    return response.status(200).send({error: ""});
+});
+
+//Retrieve documents for all friends on a user's friend list
+//Incoming: user's object _id
+//Outgoing: array of friend documents called friends
+router.get("/retrieveFriends", async (request, response) => {
```

Comment:
`/get/` or `/refresh/`?

Reply:
Not sure what this means, but this should be a get request, as you are getting all of a user's friends

Reply:
Oh sorry, I meant would `/friends/get` or `/friends/refresh` be a better route name?

Reply:
friends/get could work